lib/ash_authentication/strategies/password/actions.ex

```diff
@@ -33,12 +33,31 @@ defmodule AshAuthentication.Strategy.Password.Actions do
 
     strategy.resource
     |> Query.new()
+    |> Query.ensure_selected(List.wrap(strategy.require_confirmed_with))
     |> Query.set_context(context)
     |> Query.for_read(strategy.sign_in_action_name, params, options)
     |> Ash.read()
     |> case do
       {:ok, [user]} ->
-        {:ok, user}
+        case strategy.require_confirmed_with do
+          nil ->
+            {:ok, user}
+
+          field ->
+            if Map.get(user, field) != nil do
```

Comment:
I am not entirely sure about this, probably we should explicitly refer to `:confirmed_at` field ?

Reply:
Given on the fact that users can have multiple confirmation strategies, this comment of mine is useless and can be ignored.